### PR TITLE
Add helm chart test for Alerts plugin

### LINKED ISSUE #359: [alerts] Create chart integration tests
## Context

The alerts plugin deploys the Prometheus Alertmanager and its configuration. It also serves as a backend for the Supernova UI, which expects a valid, accessible API endpoint. The chart tests should therefore cover the following areas: 

## Tasks
- [x] The prometheus-operator CRDs are installed (if the alerts plugin manages them)
- [x] All hooks are created (Issuer, Certificate, Secrets) 
   - https://github.com/cloudoperators/greenhouse-extensions/blob/main/alerts/charts/templates/ca-secret-issuer-cert.yaml
- [x] The Alertmanager pod is operational
- [x] The Alertmanager config is valid. Reflected on the status type "Reconciled". (alertmanager object `.status`)
   ```bash
      ⎈ k get alertmanager -o yaml | yq .items[].status.conditions                 
                                                                          
          - lastTransitionTime: "2024-05-06T19:29:48Z"
            message: ""
            observedGeneration: 1
            reason: ""
            status: "True"
            type: Available
          - lastTransitionTime: "2024-05-06T19:29:48Z"
            message: 'provision alertmanager configuration: failed to initialize from secret: group_interval cannot be zero'
            observedGeneration: 1
            reason: ReconciliationFailed
            status: "False"
            type: Reconciled
   ```
   ---
   ```bash
     ⎈ k get alertmanager -o yaml | yq .items[].status.conditions[].status     
                                                                        
          True
          False
   ```
- [ ] The host created for Supernova UI is reachable and delivers an `HTTP 200` code on `/api/v2/alerts`
   - use the host from Ingress
   - the Secret `tls-{{ $.Release.Namespace }}-prometheus-auth` can be used to authenticate
   - prior art: https://github.com/cloudoperators/greenhouse-extensions/blob/3cf431b66a29c5ff2bff39ee89d4844a3a99a6f8/plutono/charts/templates/tests/test-configmap.yaml#L17-L22

### PR DESCRIPTION
**Pull request by ibakshay**
This PR, fixes #359

### CODE REVIEW

File: alerts/charts/templates/tests/test-alerts-config.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        verify "there is 1 customresourcedefinition named 'alertmanagers'"
+        verify "there is 1 customresourcedefinition named 'podmonitors'"
+        verify "there is 1 customresourcedefinition named 'probes'"
+        verify "there is 1 customresourcedefinition named 'prometheuses'"
+        verify "there is 1 customresourcedefinition named 'prometheusrules'"
+        verify "there is 1 customresourcedefinition named 'prometheusagents'"
+        verify "there is 1 customresourcedefinition named 'servicemonitors'"
+        verify "there is 1 customresourcedefinition named 'scrapeconfigs'"
+        verify "there is 1 customresourcedefinition named 'thanosrulers'"
+    }
```
Comment (richardtief) [inline alerts/charts/templates/tests/test-alerts-config.yaml:60]: This only applies if https://github.com/cloudoperators/greenhouse-extensions/blob/4330b4c18097e0d09b2b95e09373b53a30cf8b97/alerts/charts/values.yaml#L60 is set to true, right?

File: alerts/charts/templates/tests/test-alerts-config.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    {{- if .Values.alerts.alertmanager.ingress.enabled }}
+    @test "Verify that secret '{{ .Release.Namespace }}-monitoring-ca' is created when ingress is enabled" {
+        verify "there is 1 secret named '{{ .Release.Namespace }}-monitoring-ca'"
+    }
+    {{- end -}}
+
+    {{- if .Values.alerts.alertmanager.ingress.enabled }}
+    {{ if .Capabilities.APIVersions.Has "cert-manager.io/v1" }}
```
Comment (richardtief) [inline alerts/charts/templates/tests/test-alerts-config.yaml:41]:
```suggestion
    {{- if .Capabilities.APIVersions.Has "cert-manager.io/v1" }}
```

File: alerts/ci/test-values.yaml [hunk 3]
```diff
@@ -0,0 +1,225 @@
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
```
[the remark this answers is not in the document]
Comment (ibakshay) [reply alerts/ci/test-values.yaml:1]: thank you. Great find. I have fixed it now.

File: alerts/ci/test-values.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+alerts:
+  alertmanager:
+
+    ingress:
+      enabled: true
+      hosts:
+        - dummy-host
+
+
+
```
Comment (richardtief) [inline alerts/ci/test-values.yaml:16]: formatting
```suggestion
# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
# SPDX-License-Identifier: Apache-2.0
alerts:
  alertmanager:
    ingress:
      enabled: true
      hosts:
        - dummy-host
```

**Review by richardtief — CHANGES_REQUESTED** [review]
One small thing.

File: alerts/charts/Chart.yaml [hunk 5]
```diff
@@ -8,7 +8,7 @@ maintainers:
 name: alerts
 sources:
   - https://github.com/cloudoperators/greenhouse-extensions
-version: 0.12.4
+version: 0.12.5
```
Comment (richardtief) [inline alerts/charts/Chart.yaml:11]:
```suggestion
version: 0.13.0
```

File: alerts/plugindefinition.yaml [hunk 6]
```diff
@@ -6,7 +6,7 @@ kind: PluginDefinition
 metadata:
   name: alerts
 spec:
-  version: 2.0.3
+  version: 2.0.4
```
Comment (richardtief) [inline alerts/plugindefinition.yaml:9]:
```suggestion
  version: 2.1.0
```

File: alerts/plugindefinition.yaml [hunk 7]
```diff
@@ -15,7 +15,7 @@ spec:
   helmChart:
     name: alerts
     repository: oci://ghcr.io/cloudoperators/greenhouse-extensions/charts
-    version: 0.12.4
+    version: 0.12.5
```
Comment (richardtief) [inline alerts/plugindefinition.yaml:18]:
```suggestion
    version: 0.13.0
```

**Review by richardtief — CHANGES_REQUESTED** [review]
One last thing. With tests, it is a new minor version.

**Review by richardtief — APPROVED** [review]
Thanks for creating the tests!

**Pull request merged**